lib/tbot/tbot.go

```diff
@@ -224,7 +225,15 @@ func (b *Bot) Run(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 
-	if err := b.initialize(ctx); err != nil {
+	unlock, err := b.initialize(ctx)
+	defer func() {
+		if unlock != nil {
```

Comment:
if you moved the defer after the error check, you should not need this nil check

Reply:
If I move the defer after the error check, there's a possibility a lock will not be released if another error occurs inside `initialize`.